server/src/main/java/org/apache/druid/client/selector/ServerSelector.java

```diff
@@ -50,7 +51,7 @@ public ServerSelector(
       TierSelectorStrategy strategy
   )
   {
-    this.segment = new AtomicReference<>(segment);
+    this.segment = new AtomicReference<>(DataSegmentInterner.intern(segment));
```

Comment:
I'm not sure `segment` still needs to be an `AtomicReference`.. Let's figure it out in #6952.

Reply:
okay